proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+* Status: **Awaiting Review**
+* Impacted Platforms: Android
+
+## Introduction
+
+This proposal is for synching up the Android SDL library with the iOS library in terms of developer facing APIs. It accomplishes this by introducing a high level developer interface for the Android library through managers. It handles many complex aspects of SDL development, including managing files and setting information on the screen. It also creates a system in which other managers can be easily added in the future to handle tasks. Many aspects of this system are mirrored from the iOS manager system and are designed to be as close as possible to that API while maintaining common Java and Android paradigms. 
+
+## Motivation
+
+In order to work with SDL, app developers are required to learn a new API that includes low level aspects rather than a familiar, native API. The steps to accomplish this include but are not limited to:
+
+   - Read Android documentation, guides, and best practices 
+   - Implement the proxy lifecycle
+   - Manually handling the majority of RPC notifications, requests and responses
+   - Manually managing concurrent and sequential operations (e.g. image upload before on screen show)
+
+The learning curve for an app developer to understand SDL for Android is high and the knowledge is specific to the platform. Development time becomes increased by the necessary boiler plate logic needed to perform common actions. This also introduces a higher probability of error prone code. This proposal aims to fix that by handling much of the complex logic through managers which.
+
+## Proposed Solution
+
+This proposal is about adding a new manager layer to the SDL Android library. As stated, many of the managers use similar APIs to already implemented iOS APIs. It has been designed to be only a minor version change, and will not modify current public APIs.
+
+The new manager system is managed by a class called `SdlManager`. The `SdlManager` handles the lifecycle logic as well as the instantiation and disposal of submanagers. Submanagers extend `BaseSubManager` which contains mostly lifecycle logic and common methods for now, but could easily be expanded in the future.
+
+##### Intended Goals:
+1. Match the iOS library manager API as close as possible
+2. Reduce the time that it takes for developers to learn SDL for Android
+3. Reduce bad practices in system implementations
+4. Create a system that is abstracted from low level API details that will allow for more dynamic changes.
+
+### Managers
+
+##### SdlManager
+
+The SDL manager will be the main, developer-facing class that will be instantiated to handle lifecycles, submanagers, and any low level API needs. It contains a builder, an ISdl instantiation, and the proxy bridge interface. All submanagers will be created upon instantiation of this manager and their lifecycle controlled by it. 
```

Comment:
This should be "developer facing" to match other references throughout the proposal.